## 50 exercise to practice for loop
* A for loop in Python is a control flow statement that allows you to iterate over a sequence of elements, such as a list, tuple, string, or range of numbers. It executes a block of code repeatedly for each item in the sequence**.
**

Here's the basic syntax of a for loop in Ponyth**

**Synt**ax:** 

for item in sequen**

**item:** is a variable that represents each element in the sequence during each iteration.  
**sequence:** is the collection of items over which the loop iterates.




1. Print numbers from 1 to 10.
2. Print even numbers from 1 to 20.
3. Print odd numbers from 1 to 20.
4. Calculate the sum of numbers from 1 to 100.
5. Print the multiplication table of a givn number.

6. Calculate the factorial of a given number.
7. Print Fibonacci series up to a certain number of terms.
8. Print all prime numbers within a given range.
9. Count the number of vowels in a string.
10. Print the reverse of a string.
11. Print the ASCII value of each character in a string.
12. Print all uppercase letters.
13. Print all lowercase letters.
14. Print the square of numbers from 1 to 10.
15. Print the cube of numbers from 1 to 10.
16. Find the maximum number in a list.
17. Find the minimum number in a list.
18. Count the number of elements in a list.
19. Calculate the average of numbers in a list.
20. Print the elements of a list in reverse order.
21. Print the common elements from two lists.
22. Remove duplicates from a list.
23. Print the index of a specific element in a list.
24. Merge two lists into a single list.
25. Print a pattern of stars in a pyramid shape.
26. Check if a given number is a palindrome.
27. Check if a given string is a palindrome.
28. Check if a given number is Armstrong number.
29. .Count the number of digits in a given number.
30. Calculate the sum of digits in a given number.
31. Find the largest digit in a given number.
32. Find the smallest digit in a given number.
33. Convert decimal to binary.
34. Convert decimal to hexadecimal.
35. Convert decimal to octal.
36. Convert binary to decimal.
37. Convert binary to hexadecimal.
38. Convert binary to octal.
39. Convert hexadecimal to decimal.
40. Convert hexadecimal to binary.
41. Convert hexadecimal to octal.
42. Convert octal to decimal.
43. Convert octal to binary.
44. Convert octal to hexadecimal.
45. Find the factorial of numbers in a list.
46. Calculate the sum of squares of numbers in a list.
47. Find the length of each word in a sentence.
48. Count the number of occurrences of each character in a string.
49. Remove all whitespace characters from a string.
50 Check if a given number is a perfect number.r.

In [5]:
# 1. Print numbers from 1 to 10.


for i in range(11):
    print(i)

0
1
2
3
4
5
6
7
8
9
10


In [11]:
# 2. Print even numbers from 1 to 20.

for i in range(20):
    if i%2 == 0:
        print(i)
    else:
        continue
    

0
2
4
6
8
10
12
14
16
18


In [12]:
# 3. Print odd numbers from 1 to 20.

for i in range(20):
    if i%2 != 0:
        print(i)
    else:
        continue

1
3
5
7
9
11
13
15
17
19


In [20]:
# 4. Calculate the sum of numbers from 1 to 100.

num = 0
for i in range(101):
    num = num + i

print(num)

5050


In [27]:
#5. Print the multiplication table of a given number

entry = input("give a number: ")

for i in range(1,11):
    i = int(entry) * i
    print(i)
    

give a number:  18


18
36
54
72
90
108
126
144
162
180


In [18]:
# 6. Calculate the factorial of a given number.

num = input("enter a positive number:")
num1 = int(num) + 1

a = 1
for i in range(1, num1):
    a = a * i

print(a)

enter a positive number: 12


479001600


The Fibonacci series is a sequence of numbers where each number is the sum of the two preceding ones, usually starting with 0 and 1.

F(n) = F(n-1) + F(n-2)

# 7. Print Fibonacci series up to a certain number of terms.

 
The Fibonacci sequence was developed by the Italian mathematician, Leonardo Fibonacci,   
in the 13th century. The sequence of numbers, starting with zero and one,   
is a steadily increasing series where each number is equal to the sum of the preceding two numbers.  

**Each number is equal to the sum of the preceding two numbers**.  
For example, 0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377.

**Formula** =  Fn = Fn-1 + Fn-2......

f1 , f2, f3, ......fn



In [19]:
def fib(n):
    a = 0 
    b = 1

    if n == 1:
        print(a)
    else:
        print(a)
        print(b)
        
        for i in range(2,n):
            c = a + b
            a = b
            b = c
            print(c)

In [20]:
fib(5)


0
1
1
2
3


In [19]:
# 8. Print all prime numbers within a given range.
        # prime number: a whole number greater than 1 that cannot be exactly divided by any whole number other than itself and 1
num = int(input("Enter an number"))

for x in range(num+1):
    if x > 1:
        for i in range(2,x):
            if(num %i) == 0:
                break
            else:
                print(num)
            


Enter an number 25


25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
